video/out/vo_drm.c

```diff
@@ -325,6 +345,158 @@ static int modeset_prepare_dev(struct vo *vo, int fd, int conn_id,
 
 
 
+static int vt_switcher_pipe[2];
+
+static void vt_switcher_sighandler(int sig)
+{
+    unsigned char event = sig == SIGUSR1 ? EVT_RELEASE : EVT_ACQUIRE;
+    write(vt_switcher_pipe[1], &event, sizeof(event));
+}
+
+static void vt_switcher_interrupt(struct vt_switcher *s)
+{
+    unsigned char event = EVT_INTERRUPT;
+    write(vt_switcher_pipe[1], &event, sizeof(event));
+}
+
+static int vt_switcher_init(struct vt_switcher *s, struct vo *vo)
+{
+    s->vo = vo;
+    if (pipe(vt_switcher_pipe)) {
+        MP_ERR(vo, "Creating pipe failed: %s", mp_strerror(errno));
+        return -1;
+    }
+
+    s->tty_fd = open("/dev/tty", O_RDWR);
+    if (s->tty_fd < 0) {
+        MP_ERR(vo, "Can't open TTY for VT control: %s", mp_strerror(errno));
+        return -1;
+    }
+
+    struct sigaction act;
+    act.sa_handler = vt_switcher_sighandler;
+    act.sa_flags = SA_NODEFER;
+    sigemptyset(&act.sa_mask);
+    sigaction(SIGUSR1, &act, 0);
+    sigaction(SIGUSR2, &act, 0);
+
+    struct vt_mode vt_mode;
+    if (ioctl(s->tty_fd, VT_GETMODE, &vt_mode) < 0) {
+        MP_ERR(vo, "VT_GETMODE failed: %s", mp_strerror(errno));
+        return -1;
+    }
+
+    vt_mode.mode = VT_PROCESS;
+    vt_mode.relsig = SIGUSR1;
+    vt_mode.acqsig = SIGUSR2;
+    if (ioctl(s->tty_fd, VT_SETMODE, &vt_mode) < 0) {
+        MP_ERR(vo, "VT_SETMODE failed: %s", mp_strerror(errno));
+        return -1;
+    }
+
+    return 0;
+}
+
+static void vt_switcher_destroy(struct vt_switcher *s)
+{
+    close(s->tty_fd);
+    close(vt_switcher_pipe[0]);
+    close(vt_switcher_pipe[1]);
+}
```

Comment:
If you call this unconditionally, you must be sure the values are not valid FDs if initalization failed. Otherwise, you could close FD 0 (stdin).